### PR TITLE
21 implement logic for interactive example

### PR DESCRIPTION
**Pull request by IgorKozlovsky**

### CODE REVIEW

File: src/components/BoolSwitcher/types.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { BoolSwitcherNames } from 'components/BoolSwitcher/constants'
+import { ControlPropsWithName } from 'types'
+
+export type BoolSwitcherProps = UseBoolSwitcherProps & {
+  title: string
+  buttons: { title: string; value: boolean }[]
```
Comment (Lulca) [inline src/components/BoolSwitcher/types.ts:6]: create separate type of { title: string; value: boolean }

File: src/components/OptionsInput/type.ts [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+export type InputOptionsProps = ControlProps &
+  InputOptionsProp & {
+    title: string
+    name: OptionsInputNames
+  }
+
+export type UseInputOptionsProps = ControlProps &
+  InputOptionsProp & {
+    name: OptionsInputNames
+  }
```
Comment (Lulca) [inline src/components/OptionsInput/type.ts:14]: DRY create a separate type for name: OptionsInputNames and reuse

File: src/components/Example/index.tsx [hunk 3]
```diff
@@ -12,28 +11,44 @@ import {
...
     autoplaySpeed,
     animationSpeed,
-    name
+    slidesNumber,
+    spaceBetweenSlides,
+    configSlidesNumber = 3,
+    configSpaceBetweenSlides,
+    maxScreenWidth = 1200,
+    dotColor = '#d9d9d9',
+    activeDotColor = '#b70808',
```
Comment (alex-dishen) [inline src/components/Example/index.tsx:38]: Use color values from the theme

File: src/components/NumberInput/types.ts [hunk 4]
```diff
@@ -1,5 +1,9 @@
-export type InputNumberProps = {
+import { ControlPropsWithName } from 'types'
+import { NumberInputNames } from 'components/NumberInput/constants'
+
+export type UseInputNumberProps = ControlPropsWithName<number, NumberInputNames>
+
+export type InputNumberProps = UseInputNumberProps & {
   title: string
   unit?: string
```
Comment (alex-dishen) [inline src/components/NumberInput/types.ts:8]: Create a separate type for those two

File: src/components/NumberInput/useInputNumber.ts [hunk 5]
```diff
@@ -1,15 +1,24 @@
-import { ChangeEvent, useState } from 'react'
+import { useController } from 'react-hook-form'
+import { UseInputNumberProps } from './types'
 
-export const useInputNumber = (defaultValue: number) => {
-  const [number, setNumber] = useState<number>(defaultValue)
+export const useInputNumber = ({
```
Comment (alex-dishen) [inline src/components/NumberInput/useInputNumber.ts:4]: Wrong word sequence useInputNumber => useNumberInput
Do the same for file name

**Review by alex-dishen — APPROVED** [review]

File: src/components/Example/helpers.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import { CustomDot } from 'constants'
+import { StyledCustomButton } from 'components/Example/styles'
+
+export const evaluateCodeFunctionOrReturnCode = (example: {
+  code: CodeProp
+}): string => {
+  return isFunction(example.code) ? example.code() : example.code
+}
+
+export const getCustomDot = (type?: CustomDot): JSX.Element | undefined => {
```
Comment (midstem) [inline src/components/Example/helpers.tsx:12]: JSX.Element | null 

we return either something or nothing

File: src/constants.ts [hunk 7]
```diff
@@ -16,3 +16,23 @@ export enum NavigationLinks {
...
+  FADE_IN = 'fade-in'
+}
+
+export enum CustomDot {
+  MARK = 'Mark',
+  SQUARE = 'Square',
+  DEFAULT = 'Default'
+}
+
+export enum Units {
```
Comment (midstem) [inline src/constants.ts:36]: ?, for what

File: src/features/InteractiveExample/helpers.ts [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  codeString: string
+): string => {
+  if (condition) {
+    return code + codeString
+  }
+
+  return code
+}
+
+export const generateCodeParts = (codeParts: {condition: unknown, codeString: string}[]): string => {
```
Comment (midstem) [inline src/features/InteractiveExample/helpers.ts:17]: bring out the type

File: src/features/InteractiveExample/helpers.ts [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+  let code = ''
+
+  for (const part of codeParts) {
```
Comment (midstem) [inline src/features/InteractiveExample/helpers.ts:20]: why are you don`t use forEach?

**Pull request merged**